### PR TITLE
Putting it all together!

### PR DESCRIPTION
**Pull request by psiemens**
I merged code from Mark, Jason, and Tyler in an effort to tie everything together for our launch. Here's what I changed:

- Tweaked CSS styles to better match the mockup
- Moved the chart script to a separate file (chart.js) and got rid of `chart2.html`
- Made some small changes to `index.js` to make sure all of the AJAX calls were working properly

Everything is looking really good! Great work guys 😄

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ var ENDPOINTS = {
   }
 };
 
+function numberWithCommas(x) {
+  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (JimFung) [inline js/index.js:29]: Please leave a comment here for what this regex does, not immediately obvious at first glance.

File: js/chart.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    if((result[i][0] == (hour-1)) && (result[i][1] >= counter)) {
+      x.push(Number(result[i][2]));
+      counter++;
+    }
+  }
+  // filtered array in past hour
+  return x;
+}
+
+var request = $.ajax({
```
Comment (JimFung) [inline js/chart.js:39]: Don't need to save the ajax object if you are going to define the success event in the ajax definition.

File: js/chart.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  type: 'GET',
+  url: ENDPOINTS.pageviews.hour,
+  dataType: 'jsonp',
+  //This only triggers if the ajax call was successful
+  //TODO: Define failure case
+  success: function(data) {
+    drawGraph(data);
```
Comment (blx) [inline js/chart.js:46]: this can be pointfree:
```js
success: drawGraph

File: js/chart.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+  }
+});
+
+function drawGraph(reply){
+  console.log('Here is the data from the ajax call')
```
Comment (blx) [inline js/chart.js:51]: log statements necessary?

File: js/chart.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+      .ylabel("Page Views");
+
+  var svg = d3.select("#pageviews-chart").append("svg")
+      .datum(data)
+      .call(xy_chart) ;
+
+  function d3_xy_chart() {
+      var width = width,
+          height = height,
+          xlabel = "X Axis Label",
```
Comment (blx) [inline js/chart.js:85]: is this a useful label?

File: stylesheet.css [hunk 6]
```diff
@@ -1,118 +1,200 @@
...
+
+header {
+  display: flex;
+  opacity: 0.7;
+  font-size: 1.5rem;
+  font-weight: 100;
+  padding-left: 2rem;
+  padding-right: 2rem;
+  padding-top: 2rem;
+  padding-bottom: 2rem;
```
Comment (blx) [inline stylesheet.css:27]:
```css
padding: 2rem;
```
? or is that against house style

File: js/chart.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
+// Filter out an array with the users in past 60 minutes
+function filter(data) {
+  //this gives you an array of tuples where the first element is the hour and the last element is the # of users
+  var result = data.rows.map(
+    function(elem) {
+      return [elem[0], elem[1], elem[2]]
```
Comment (blx) [inline js/chart.js:6]: given that this project has only two js files, it would be ideal if they both either use semicolons or don't

File: js/chart.js [hunk 8]
```diff
@@ -0,0 +1,220 @@
+// Filter out an array with the users in past 60 minutes
+function filter(data) {
```
Comment (blx) [inline js/chart.js:2]: i am just a rando person, but strongly advise against calling anything `filter`. That is not a useful name. Call this `filterUsersPastHour` etc and the name will be more clear plus the comment will not be required.

File: js/chart.js [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+                 x: xData,
+                 y: yData }, ] ;
+
+  var xy_chart = d3_xy_chart()
+      .width(width)
+      .height(height)
+      .xlabel("Time")
+      .ylabel("Page Views");
+
+  var svg = d3.select("#pageviews-chart").append("svg")
```
Comment (blx) [inline js/chart.js:78]: this is the third reference to this ID string, ideally it would be a constant or parameter as opposed to a fragile magic string.

File: analytics.html [hunk 10]
```diff
@@ -1,65 +1,90 @@
...
-                    <hr/>
-                    <p></p>
-                </div>
-                <div class="box"> <h3> Statistics from: </h3> <hr/> <p id="stats-time-range">Past Hour</p></div>
+    <div class="container">
+
+      <header>
+        <div id="date">Tuesday, February 14, 2017</div>
+        <div id="title">Ubyssey Analytics</div>
+        <div id="time"><span>3:04pm</span></div>
```
Comment (blx) [inline analytics.html:16]: sup with these hardcoded datetime values?

File: analytics.html [hunk 11]
```diff
@@ -1,65 +1,90 @@
...
-            <div id="date">Tuesday, February 14, 2017</div>
-            <div id="title">Ubyssey Analytics</div>
-            <div id="time"><span>3:04pm</span></div>
-        </div>
+  <head>
+    <link rel="stylesheet" type="text/css" href="stylesheet.css">
+    <script type="text/javascript" src="https://use.fontawesome.com/432212f1dd.js"></script>
+    <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js"></script>
+    <script type="text/javascript" src="http://d3js.org/d3.v3.js"></script>
```
Comment (blx) [inline analytics.html:7]: `text/javascript` can easily be excluded if desired, since browsers assume it anyways

**Comment by JimFung** [comment]
Did any of the suggested changes get made?
